File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.java
class Solution {
    public int[] runningSum(int[] nums) {
        int lastValue = 0;
        final var result = new int[nums.length];

        for (int i = 0; i < nums.length; i++) {
            final var value = nums[i];
            lastValue += value;
            result[i] = lastValue; 
        }

        return result;
    }
}

File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.swift
class Solution {
    func runningSum(_ nums: [Int]) -> [Int] {
        
        var prevSum = 0
        var results = [Int]()
        results.reserveCapacity(nums.count)

        for num in nums {

            prevSum += num
            results.append(prevSum)
        }

        return results
    }
}